### PR TITLE
Remove inheritance from object

### PR DESCRIPTION
**Pull request by forslund**
## Description
Inheriting from object isn't necessary in python3. Removing it to keep the code cleaner and more consistent.

## How to test
Make sure tests passes and mycroft still functions

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

File: mycroft/api/__init__.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ class InternetDown(RequestException):
     pass
 
 
-class Api(object):
-    """ Generic object to wrap web APIs """
+class Api():
```
Comment (MatthewScholefield) [inline mycroft/api/__init__.py]: We might as well exclude the extra parentheses here as well (so it becomes `class Api:`).

**Comment by JarbasAl** [comment]
i would like to keep the object in there, i know we moved to python 3 and python 2 support is not a priority, but no need to make it harder if someone wants to port it

i got used to write code compatible with python2 and 3, and while support shouldn't be mandatory i think compatibility is a good habit

**Comment by JarbasAl** [comment]
this is still valid syntax in python 3, it's not "8-year old syntax", i could argue old style classes are 8-year old syntax still supported, but to be fair even the python3 docs use it 

i see it as supporting "bad" developer habits, just like enforcing pep8 is a good habit even if the code was valid otherwise
 
i totally get what your point about this not being a library though, and supporting python2 is not the main point i wanted to make

File: mycroft/api/__init__.py (hunk 1 above)
Comment (forslund) [reply mycroft/api/__init__.py]: Ah, right you are!

Should be fixed now.

**Comment by forslund** [comment]
I certainly agree that you have a valid point @JarbasAI but I think keeping a "clean" (for some value of clean) code base is more important for this project. I believe this increases readability while no functionality is lost:

- Explicitly Inheriting from object doesn't add any relevant information.
- Classes that actually inherits from something of interest is easier to locate in the code.
- Easier to maintain consistency

**Comment by penrods** [comment]
My $0.02 on this is that we did make the cut over to Python 3.  As I'm working on code, I'm eliminating Python 2-isms, such as using the simpler ```super().__init__()``` instead of ```super(ChildB, self).__init__()```.  It is just more readable.  Unless we are actively forcing ourselves to be 2.x compatible, there is no reason to be kinda-compatible.  A semi-compatible hybrid is just ugly and really less useful to both the 3.x (who have to suffer old syntax) and the 2.x (who pull in something, but can't actually get it to work).

**Pull request merged**